discord/ext/commands/parameters.py

```diff
@@ -0,0 +1,264 @@
+        return self.default
+
+
+def parameter(
+    *,
+    converter: Any = empty,
+    default: Any = empty,
+    displayed_default: str = empty,
+) -> Any:
+    r"""parameter(\*, converter=..., default=..., displayed_default=...)
+
+    A way to assign custom metadata for a :class:`Command`\'s parameter.
+
+    Examples
+    --------
+
+    A custom converter, whilst annotating a parameter with a custom converter works at runtime, type checkers don't like it
+    cause they can't understand what's going on.
+
+    .. code-block:: python3
+
+        class SomeType:
+            foo: int
+
+        class MyVeryCoolConverter(commands.Converter[SomeType]):
+            ...  # implementation left as an exercise for the reader
+
+        @bot.command()
+        async def bar(ctx, cool_value: MyVeryCoolConverter):
+            cool_value.foo  # type checker warns MyVeryCoolConverter has no value foo (uh-oh)
+
+    However, fear not we can use :func:`parameter` to tell type checkers what's going on.
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def bar(ctx, cool_value: SomeType = commands.parameter(converter=MyVeryCoolConverter)):
+            cool_value.foo  # no error (hurray)
+
+    A custom default can be used to have late binding behaviour
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def wave(to: discord.User = commands.parameter(default=lambda ctx: ctx.author)):
+            await ctx.send(f'Hello {to.mention} :wave:')
+
+    Because this is such a common use-case, the library provides :obj:`.Author`, :obj:`.CurrentChannel` and
+    :obj:`.CurrentGuild`, armed with this we can simplify ``wave`` to:
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def wave(to: discord.User = commands.Author):
+            await ctx.send(f'Hello {to.mention} :wave:')
+
+    :obj:`.Author` and co also have other benefits like having the displayed default being filled.
+
+    .. versionadded:: 2.0.0
+
+    Parameters
+    ----------
+    converter: Any
+        The converter to use for this parameter, this replaces the annotation at runtime which is transparent to type checkers.
+    default: Any
+        The default value for the parameter, if this is a :term:`callable` or a |coroutine_link|_ it is called with a
+        positional :class:`Context` argument.
+    displayed_default: :class:`str`
+        The displayed default in :attr:`Command.signature`.
+    """
+    return Parameter(
+        name="empty",
+        kind=inspect.Parameter.POSITIONAL_OR_KEYWORD,
+        annotation=converter,
+        default=default,
+        displayed_default=displayed_default,
+    )
+
+
+param = parameter
+"""param(*, converter=..., default=..., displayed_default=...)
+
+An alias for :func:`parameter`.
+"""
+
+# some handy defaults
+Author: Union[Member, User] = parameter(
+    default=attrgetter("author"),
+    displayed_default="<you>",
```

Comment:
Shouldn't this be `[you]` since the presence of the default makes this optional?

Reply:
I thought it should be <> as it's meant to represent a custom object you can't construct similar to https://discordpy.readthedocs.io/en/master/ext/commands/api.html#discord.ext.commands.Context.send_help and others, interested as to what @Rapptz thinks.

Reply:
Never mind, I got confused a bit... The usage ends up being `[p]command [argument=<you>]`, I don't know why I thought it would be `[p]command <you>`. Sorry about that 😄